File: solutions/php/rotational-cipher/1/RotationalCipher.php
<?php
declare(strict_types=1);

class RotationalCipher
{
    function rotate(string $text, int $shift): string
    {
        $result = '';
        for ($i = 0; $i < strlen($text); $i++) {
            $char = $text[$i];
            if (ctype_alpha($char)) {
                $offset = ord(strtolower($char)) - ord('a');
                $rotatedChar = chr(ord('a') + (($offset + $shift) % 26));
                $result .=
                    $char === strtoupper($char)
                        ? strtoupper($rotatedChar)
                        : $rotatedChar;
            } else {
                $result .= $char;
            }
        }
        return $result;
    }
}

File: solutions/lua/rotational-cipher/1/rotational-cipher.lua
return {
    rotate = function(input, key)
        local result = ""
        for i = 1, #input do
            local c = input:sub(i, i)
            if c >= 'A' and c <= 'Z' then
                result = result .. string.char((c:byte() - 65 + key) % 26 + 65)
            elseif c >= 'a' and c <= 'z' then
                result = result .. string.char((c:byte() - 97 + key) % 26 + 97)
            else
                result = result .. c
            end
        end
        return result
    end,
}
